main.nf

```diff
@@ -1,84 +1,396 @@
+                                                    the parameter below --rna_classes_to_replace_host can help to summarise these)
+                                                  (Default: false)
+      --rna_classes_to_replace_host      [file]   Located within the data/ directory, this tab delimited file contains headers which 
+                                                  groups similar types of RNA classes together. This helps to keep the RNA-class 
+                                                  names simplified for plotting purposes.
+                                                  (Default: $projectDir/data/RNA_classes_to_replace.tsv)
+   
+    Report options:
+      --email                   [email]   Set this parameter to your e-mail address to get a summary e-mail with details of the 
+                                          run sent to you when the workflow exits
+                                          (Default: false)
+      --email_on_fail           [email]   Same as --email, except only send mail if the workflow is not successful
+                                          (Default: false)  
+      --max_multiqc_email_size  [str]     Theshold size for MultiQC report to be attached in notification email. 
+                                          If file generated by pipeline exceeds the threshold, it will not be attached
+                                          (Default: 25MB)
+      --plaintext_email         [bool]    Set to receive plain-text e-mails instead of HTML formatted
+                                          (Default: false)
+      --monochrome_logs         [bool]    Set to disable colourful command line output and live life in monochrome
+                                          (Default: false)
+      --multiqc_config          [bool]    Specify path to a custom MultiQC configuration file.
+                                          (Default: false)
+
     AWSBatch options:
       --awsqueue [str]                The AWSBatch JobQueue that needs to be set when running on AWSBatch
       --awsregion [str]               The AWS Region for your AWS Batch job to run on
       --awscli [str]                  Path to the AWS CLI tool
     """.stripIndent()
 }
 
+
 // Show help message
 if (params.help) {
     helpMessage()
     exit 0
 }
 
+
+
 /*
- * SET UP CONFIGURATION VARIABLES
+--------------------------------------------------------------------
+
+SET UP CONFIGURATIONs AND IDENTIFY USER-SPECIFIED VARIABLES
+
+--------------------------------------------------------------------
  */
 
+//----------
 // Check if genome exists in the config file
+//----------
 if (params.genomes && params.genome && !params.genomes.containsKey(params.genome)) {
     exit 1, "The provided genome '${params.genome}' is not available in the iGenomes file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
```

Comment:
You should actually test the existence of `params.genome_host` and `params.genome_pathogen` and if `params.genomes` contains keys from `params.genome_host` and `params.genome_pathogen`.

Reply:
@bozmik I'll leave this one up to you

Reply:
I assumed the genomes are stored in the genomes.config instead of iGenomes file.

`if (params.genomes && params.genome_host && !params.genomes.containsKey(params.genome_host)) {
    exit 1, "The provided genome '${params.genome_host}' is not available in the genomes.config file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
}

if (params.genomes && params.genome_pathogen && !params.genomes.containsKey(params.genome_pathogen)) {
    exit 1, "The provided genome '${params.genome_pathogen}' is not available in the genomes.config file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
}
`